[wasm-mt] PInvokeTableGenerator doesn't like the wasm-mt-eventpipe sample

Build the runtime with `/p:WasmEnableThreads=true`, try to build the wasm-mt-eventpipe wasm sample with AOT compilation:

```
src/mono/wasm/build/WasmApp.Native.targets(276,5): error MSB4018: (NETCORE_ENGINEERING_TELEMETRY=Build) The "PInvokeTableGenerator" task failed unexpectedly.
System.IO.FileLoadException: The assembly 'System.Runtime, Version=7.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' has already loaded been loaded into this MetadataLoadContext.
   at System.Reflection.MetadataLoadContext.LoadFromStreamCore(Stream peStream)
   at PInvokeTableGenerator.GenPInvokeTable(String[] pinvokeModules, String[] assemblies) in /_/src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs:line 68
   at PInvokeTableGenerator.Execute() in /_/src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs:line 45
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask)
```

We're probably pulling in a ref assembly into the build.

--- a/src/libraries/System.Threading.ThreadPool.WebAssembly.Threading/ref/System.Threading.ThreadPool.WebAssembly.Threading.csproj
+++ b/src/libraries/System.Threading.ThreadPool.WebAssembly.Threading/ref/System.Threading.ThreadPool.WebAssembly.Threading.csproj
@@ -16,7 +16,7 @@
     <Compile Include="$(LibrariesProjectRoot)System.Threading.ThreadPool\ref\System.Threading.ThreadPool.cs" />
   </ItemGroup>
   <ItemGroup>
-    <ProjectReference Include="..\..\System.Runtime\ref\System.Runtime.csproj" />
-    <ProjectReference Include="..\..\System.Threading.Overlapped\ref\System.Threading.Overlapped.csproj" />
+    <ProjectReference Include="..\..\System.Runtime\ref\System.Runtime.csproj" PrivateAssets="all" />
+    <ProjectReference Include="..\..\System.Threading.Overlapped\ref\System.Threading.Overlapped.csproj" PrivateAssets="all" />
   </ItemGroup>
 </Project>

